lib/compress/zstdmt_compress.c

```diff
@@ -1239,174 +1174,6 @@ static size_t ZSTDMT_computeOverlapSize(const ZSTD_CCtx_params* params)
     return (ovLog==0) ? 0 : (size_t)1 << ovLog;
 }
 
-static unsigned
-ZSTDMT_computeNbJobs(const ZSTD_CCtx_params* params, size_t srcSize, unsigned nbWorkers)
-{
-    assert(nbWorkers>0);
-    {   size_t const jobSizeTarget = (size_t)1 << ZSTDMT_computeTargetJobLog(params);
-        size_t const jobMaxSize = jobSizeTarget << 2;
-        size_t const passSizeMax = jobMaxSize * nbWorkers;
-        unsigned const multiplier = (unsigned)(srcSize / passSizeMax) + 1;
-        unsigned const nbJobsLarge = multiplier * nbWorkers;
-        unsigned const nbJobsMax = (unsigned)(srcSize / jobSizeTarget) + 1;
-        unsigned const nbJobsSmall = MIN(nbJobsMax, nbWorkers);
-        return (multiplier>1) ? nbJobsLarge : nbJobsSmall;
-}   }
-
-/* ZSTDMT_compress_advanced_internal() :
- * This is a blocking function : it will only give back control to caller after finishing its compression job.
- */
-static size_t
-ZSTDMT_compress_advanced_internal(
```

Comment:
Early in the life of `zstdmt`, one of the benefits of this function
is that it was able to compress directly from `src` into `dst`
provided there is enough `dstCapacity` (`<= ZSTD_compressBound(srcSize)`),
thus skipping the need to use intermediate buffers,
thus saving memory, and some unnecessary `memcpy()`.

With this function removed,
is the replacement able to retain this property ?

Reply:
> is the replacement able to retain this property ?

No. We already weren't saving `memory` because when using the streaming API we have already allocated all of our buffers at this point. With this change, we are doing one extra memcpy for the source buffers.